libs/langchain-mongodb/langchain_mongodb/graphrag/graph.py

```diff
@@ -122,6 +129,48 @@ def __init__(
               - If "warn", the default, documents will be inserted but errors logged.
               - If "error", an exception will be raised if any document does not match the schema.
         """
+        self._schema = deepcopy(entity_schema)
+        if connection_string and collection is not None:
+            raise ValueError(
+                "Pass one of: connection_string, database_name, and collection_name"
+                "OR a MongoDB Collection."
+            )
+        if collection is not None:
+            if not isinstance(collection, Collection):
+                raise ValueError(
+                    "collection must be a MongoDB Collection. "
+                    "Consider using connection_string, database_name, and collection_name."
+                )
+            if validate:
+                try:
+                    collection.database.command(
+                        "collMod",
+                        collection.name,
+                        validator={"$jsonSchema": self._schema},
+                        validationAction=validation_action,
+                    )
+                except OperationFailure:
+                    logger.warning(
+                        "Validation will NOT be performed. "
+                        "User must be DB Admin to add validation **after** a Collection is created. \n"
+                        "Please add validator when you create collection: "
+                        "db.create_collection.(coll_name, validator={'$jsonSchema': schema.entity_schema})"
+                    )
+        else:
+            client: MongoClient = MongoClient(
```

Comment:
We can't assume that the collection doesn't already exist just because we're creating the client.

Reply:
@blink1073 This is inside a warning message.

Reply:
My question is: if the collection already exists, what is the behavior here?

Reply:
We do *attempt*. It's in a try/except block. Let's discuss on slack if we want to delay release of a version for this.

Reply:
We do not attempt for a collection that already exists, which may have been created by the user, no?